### PR TITLE
Add throttle operation test

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:1908619,1908620,1908621,1908622,1908623,1908625,1908626,1908627,1908628,1908629,1908630

Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=throttle_operation_test --guestname=RHEL.8.4.0,Win2019 --clone=no
(01/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.bps_non_4k: PASS (323.28 s)
 (02/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.bps_iops: PASS (255.42 s)
 (03/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.to_burst: PASS (388.59 s)
 (04/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.to_empty: PASS (265.42 s)
 (05/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.group_move.to_normal: PASS (192.25 s)
 (06/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.group_move.to_burst: PASS (253.85 s)
 (07/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.with_reboot: PASS (282.08 s)
 (08/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.with_stop_resume: PASS (255.16 s)
 (09/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.with_hotplug: PASS (262.89 s)
 (10/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.negative.wrong_value: PASS (254.99 s)
 (11/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.throttle_operation_test.negative.wrong_option: PASS (253.63 s)
 (12/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.bps_non_4k: PASS (303.40 s)
 (13/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.bps_iops: PASS (260.12 s)
 (14/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.to_burst: PASS (394.16 s)
 (15/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.group_change.to_empty: PASS (263.38 s)
 (16/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.group_move.to_normal: PASS (197.51 s)
 (17/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.group_move.to_burst: PASS (258.55 s)
 (18/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.with_reboot: PASS (319.23 s)
 (19/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.with_stop_resume: PASS (259.63 s)
 (20/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.with_hotplug: PASS (267.00 s)
 (21/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.negative.wrong_value: PASS (261.75 s)
 (22/22) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.throttle_operation_test.negative.wrong_option: PASS (259.89 s)

**Comment by qingwangrh** [comment]
@YongxueHong @zhencliu Please help to review, thanks.

File: qemu/tests/cfg/throttle_operation_test.cfg [hunk 1]
```diff
@@ -0,0 +1,91 @@
+- throttle_operation_test:
+    type = throttle_operation_test
+    qemu_force_use_drive_expression = no
```
Comment (YongxueHong) [inline qemu/tests/cfg/throttle_operation_test.cfg:3]:
```suggestion
```
This default of `qemu_force_use_drive_expression` is `no` in the avocado-vt framework.

File: qemu/tests/cfg/throttle_operation_test.cfg [hunk 2]
```diff
@@ -0,0 +1,91 @@
+- throttle_operation_test:
+    type = throttle_operation_test
```
Comment (zhencliu) [inline qemu/tests/cfg/throttle_operation_test.cfg:2]: add virt_test_type = qemu ?

File: qemu/tests/throttle_operation_test.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
+"""IO-Throttling group and other operation relevant testing"""
+import json
+import logging
+
+from provider.block_devices_plug import BlockDevicesPlug
```
Comment (zhencliu) [inline qemu/tests/throttle_operation_test.py:5]: xchange line5 and line11?
order:
standard lib
virttest(vt)
provider (tp-qemu)

**Comment by zhencliu** [comment]
LGTM

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**